dbms/src/Interpreters/AsynchronousMetrics.cpp

```diff
@@ -177,11 +177,14 @@ void AsynchronousMetrics::update()
 
                 if (auto dt_storage = std::dynamic_pointer_cast<StorageDeltaMerge>(table); dt_storage)
                 {
-                    auto stat = dt_storage->getStore()->getStat();
-                    calculateMax(max_dt_stable_oldest_snapshot_lifetime, stat.storage_stable_oldest_snapshot_lifetime);
-                    calculateMax(max_dt_delta_oldest_snapshot_lifetime, stat.storage_delta_oldest_snapshot_lifetime);
-                    calculateMax(max_dt_meta_oldest_snapshot_lifetime, stat.storage_meta_oldest_snapshot_lifetime);
-                    calculateMax(max_dt_background_tasks_length, stat.background_tasks_length);
+                    if (auto store = dt_storage->getStoreIfInited(); store)
```

Comment:
```suggestion
                    if (auto store = dt_storage->getStoreIfInited())
```

Reply:
It seems we always do similar check in this way. I prefer to keep the original way to do the check.

Reply:
Using the return result of an assign statement is not recommended, it introduce bug like `if (a == b) / if (a = b)`.
Using `if (auto a = b; a)` or `if (auto a = b; a != nullptr)` is more explicit

Reply:
Get it, thank you